great_expectations/data_context/store/store_backend.py

```diff
@@ -263,6 +264,24 @@ def _has_key(self, key):
     def remove_key(self, key) -> None:
         del self._store[key]
 
+    def build_key(self, name: Optional[str], id_: Optional[str]) -> DataContextKey:
+        """
+        Utility to build a conditionally build specific key types based on input arguments.
+        Note that the implementation of this function by child classes is optional.
+
+        Args:
+            name: The name of the object to be stored
+            id_: The id of the object to be stored
+
+        Returns:
+            key: The key generated as a result of the input name and id_
+
+        Raises:
+            NotImplementedError if not defined by child classes in the StoreBackend hierarchy
```

Comment:
How about this? Nitpick so feel free to ignore.
```suggestion
            NotImplementedError if used without being overridden by child classes in the StoreBackend hierarchy
```